src/main/java/thaumicboots/api/ItemBoots.java

```diff
@@ -0,0 +1,275 @@
+        return tier;
+    }
+
+    public boolean getIsRepairable(ItemStack par1ItemStack, ItemStack par2ItemStack) {
+        return par2ItemStack.isItemEqual(new ItemStack(Items.leather))
+                || super.getIsRepairable(par1ItemStack, par2ItemStack);
+    }
+
+    protected float computeBonus(ItemStack itemStack, EntityPlayer player) {
+        int ticks = player.inventory.armorItemInSlot(0).stackTagCompound.getInteger("runTicks");
+        float bonus = runBonus + ((ticks / 5) * longrunningbonus);
+        return bonus;
+    }
+
+    @Override
+    public void onArmorTick(World world, EntityPlayer player, ItemStack itemStack) {
+        if (player.moveForward <= 0F) {
+            return;
+        }
+
+        float bonus = getSpeedModifier();
+        stepHeight(player);
+        if (steadyBonus) {
+            runningTicks(player);
+            bonus = computeBonus(itemStack, player);
+        }
+
+        bonus *= itemStack.stackTagCompound.getDouble(TAG_MODE_SPEED);
+        applyBonus(player, bonus);
+
+        if (negateFall) {
+            if (player.fallDistance > 0.0F) {
+                player.fallDistance = 0.0F;
+            }
+        }
+    }
+
+    public void stepHeight(EntityPlayer player) {
+        if (player.worldObj.isRemote) {
+            if (!Thaumcraft.instance.entityEventHandler.prevStep.containsKey(Integer.valueOf(player.getEntityId()))) {
+                Thaumcraft.instance.entityEventHandler.prevStep
+                        .put(Integer.valueOf(player.getEntityId()), Float.valueOf(player.stepHeight));
+            }
+            player.stepHeight = 1.0F;
+        }
+    }
+
+    public void runningTicks(EntityPlayer player) {
+        if (!player.inventory.armorItemInSlot(0).hasTagCompound()) {
+            NBTTagCompound par1NBTTagCompound = new NBTTagCompound();
+            player.inventory.armorItemInSlot(0).setTagCompound(par1NBTTagCompound);
+            player.inventory.armorItemInSlot(0).stackTagCompound.setInteger("runTicks", 0);
+        }
+    }
+
+    public void applyBonus(EntityPlayer player, float bonus) {
+        if (waterEffects) {
+            if (player.isInWater()) {
+                bonus /= 4.0F;
+            }
+        }
+        if (player.onGround || player.isOnLadder() || player.capabilities.isFlying) {
+            player.moveFlying(0.0F, 1.0F, bonus);
+        } else if (Hover.getHover(player.getEntityId())) {
+            player.jumpMovementFactor = 0.03F;
+        } else {
+            player.jumpMovementFactor = 0.05F;
+        }
+    }
+
+    // taken from Vazkii
+    public static ItemStack getBoots(EntityPlayer player) {
+        ItemStack stack1 = player.getCurrentArmor(0);
+        return isBoot(stack1) ? stack1 : null;
+    }
+
+    private static boolean isBoot(ItemStack stack) {
+        return stack != null && (stack.getItem() instanceof ItemBoots);
+    }
+
+    @Optional.Method(modid = "gtnhlib")
+    @SideOnly(Side.CLIENT)
```

Comment:
`@SideOnly` is not needed here

Reply:
Bizarrely, it is, I've done a lot of testing around this, it crashes the server if it's not here.